Feature/curry dictionary updates

ColmBhandal opened the pull request:
## Description

Fixed bug in the ``ContainsKeyTuple`` method

ColmBhandal committed:
Added test for false case of contains key tuple prefix

ColmBhandal committed:
Added enumerables for values and key-value pairs

ColmBhandal committed:
Added test for contains key tuple prefix on empty dictionary

ColmBhandal committed:
Renamed variable

ColmBhandal committed:
Added test to ensure contains key tuple returns false for empty dictionary

ColmBhandal committed:
Fixed bug in contains key tuple method

Calling on the prefix method now and just checking the arity beforehand

hctrdev commented on CsharpExtras/Map/Dictionary/Curry/CurryDictionaryBase.cs:
```diff
@@ -11,6 +11,11 @@ abstract class CurryDictionaryBase<TKey, TVal> : ICurryDictionary<TKey, TVal>
         public abstract NonnegativeInteger Arity { get; }
         public abstract IEnumerable<IList<TKey>> Keyset();
 
+        public IEnumerable<(IList<TKey>, TVal)> KeyValuePairs() =>
```
I think it is worth adding a couple of tests for these new methods (`KeyValuePairs` and `Values`). Just to make sure they return the expected keys and do not fail on empty dictionaries.

hctrdev commented on CsharpExtras/Map/Dictionary/Curry/ICurryDictionary.cs:
```diff
@@ -8,6 +8,8 @@ public interface ICurryDictionary<TKey, TVal>
         TVal this[params TKey[] keyTuple] { get; }
         IEnumerable<IList<TKey>> Keyset();
+        IEnumerable<(IList<TKey>, TVal)> KeyValuePairs();
+        IEnumerable<TVal> Values();
```
Should these be methods or properties? The standard C# dictionaries expose `Keys` and `Values` as properties. Is there a particular reason you made these methods instead?

ColmBhandal replied:
Yeah I originally had Keys as a property, but it felt like a lie to the user as the enumeration was walking through each individual item... However, I suppose that's the way for most enumerable properties... I'll change em to properties so

ColmBhandal committed:
Refactored Key / Value / Key-Value pairs to properties

ColmBhandal committed:
Added tests for Keys, KeyValuePairs and Values

Added tests for empty dictionary case
Added tests for checking enumerables after adding known entries
